Src/FluentAssertions/AssertionOptions.cs

```diff
@@ -18,6 +16,7 @@ public static class AssertionOptions
         static AssertionOptions()
         {
             EquivalencySteps = new EquivalencyStepCollection();
+            TaskTimeout = 30.Seconds();
```

Comment:
`30` seems like a rather arbitrary choice to me?

If I saw something like this snippet, my immediate thought is "when do I know that something does *not* complete?" and the halting problem.
```c#
something.Should().CompleteAsync()
```

I would need to consult implementation or documentation to know that there is a default timeout.
I prefer this being explicit in the code.

Just my thoughts.

Reply:
But if we include the timeout in the failure message?

Reply:
It can also be mentioned in the documentation.

Reply:
My idea was that one does not have to read a failure message or documentation to see what some default timeout is.
I think we already did a design mistake in `BeCloseTo(DateTime, int precision = 20)` by specifying a default precision. 
Was constitutes as two dates being close is context specific.

```c#
/// <summary>
/// Defined by company spec §1.2.3
/// </summary>
private static readonly TimeSpan DefaultTimeout = TimeSpan.FromSeconds(30);

MyTestMethod()
{
   subject.Should().CompleteWithinAsync(DefaultTimeout);
}
```

Reply:
I understand your thoghts, @jnyrup . I do not want to start a long discussion here.

You and @dennisdoomen, you are leading together. Please decide whether such default values are accepted or not.
And, **please**, document your design rules to prevent discussion and implementation effort for /dev/null.

For me, I do not see a problem with default behaviors. E.g. `BeEquivalentTo` has a lot of default behaviors too.

Reply:
@jnyrup what do you propose instead? Just wait indefinitely?

Reply:
Waiting indefinitely is useless. In this case you can just use `await`.
This is why I introduced this extension: I want to prevent waiting indefinitely...

Reply:
@dennisdoomen Removing the `CompleteAsync()` and only have the `CompleteWithinAsync(TimeSpan)`. 

Just as `GenericAsyncFunctionAssertions<TResult>` and `NonGenericAsyncFunctionAssertions` only have `CompleteWithin(TimeSpan)` and `CompleteWithinAsync(TimeSpan)` assertions.

@lg2de I'm very sorry that your efforts once again ends up in these exhausting discussions.
I'll start see if we can making some design guidelines. 
That said, I can't make any promises on:
* when they will be ready
* that they be fully covering
* that the *current* codebase lives up to those guidelines.

Reply:
So the general design decision is "no default timeouts". I can live with that.

Reply:
Ok, I'll remove.
Anyway I do not really understand what is the difference between
* "default behavior A": BeEquivalentTo checks all public properties
* "default behavior B": CompleteAsync uses timeout of 30 seconds